File: generated-code/php/codecraft/TcpReadWrite/Model/MoveAction.php
<?php

namespace Model {
    require_once 'Vec2Int.php';

    /**
     * Move action
     */
    class MoveAction
    {
        /**
         * Target position
         */
        public $target;
        /**
         * Whether to try find closest position, if path to target is not found
         */
        public $findClosestPosition;
        /**
         * Whether to destroy other entities on the way
         */
        public $breakThrough;
    
        function __construct($target, $findClosestPosition, $breakThrough)
        {
            $this->target = $target;
            $this->findClosestPosition = $findClosestPosition;
            $this->breakThrough = $breakThrough;
        }
    
        /**
         * Read MoveAction from input stream
         */
        public static function readFrom($stream)
        {
            $target = \Vec2Int::readFrom($stream);
            $findClosestPosition = $stream->readBool();
            $breakThrough = $stream->readBool();
            return new MoveAction($target, $findClosestPosition, $breakThrough);
        }
        
        /**
         * Write MoveAction to output stream
         */
        public function writeTo($stream)
        {
            $this->target->writeTo($stream);
            $stream->writeBool($this->findClosestPosition);
            $stream->writeBool($this->breakThrough);
        }
    }
}

File: generated-code/python/codecraft/TcpReadWrite/model/move_action.py
from vec2_int import Vec2Int

class MoveAction:
    """Move action"""

    def __init__(self, target, find_closest_position, break_through):
        self.target = target
        """Target position"""
        self.find_closest_position = find_closest_position
        """Whether to try find closest position, if path to target is not found"""
        self.break_through = break_through
        """Whether to destroy other entities on the way"""

    @staticmethod
    def read_from(stream):
        """Read MoveAction from input stream
        """
        target = Vec2Int.read_from(stream)
        find_closest_position = stream.read_bool()
        break_through = stream.read_bool()
        return MoveAction(target, find_closest_position, break_through)

    def write_to(self, stream):
        """Write MoveAction to output stream
        """
        self.target.write_to(stream)
        stream.write_bool(self.find_closest_position)
        stream.write_bool(self.break_through)

    def __repr__(self):
        return "MoveAction(" + \
            repr(self.target) + \
            ", " + \
            repr(self.find_closest_position) + \
            ", " + \
            repr(self.break_through) + \
            ")"